organisaatio-ui/src/components/Modaalit/ToimipisteenYhdistys/TSBody.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useContext } from 'react';
+import { LanguageContext } from '../../../contexts/contexts';
+import styles from './ToimipisteenYhdistys.module.css';
+import DatePickerInput from '@opetushallitus/virkailija-ui-components/DatePickerInput';
+import Select from '@opetushallitus/virkailija-ui-components/Select';
+import { Option, Organisaatio, ResolvedRakenne, SiirraOrganisaatioon } from '../../../types/types';
+import { useOrganisaatioHaku } from '../../../api/organisaatio';
+import Spin from '@opetushallitus/virkailija-ui-components/Spin';
+import { warning } from '../../Notification/Notification';
+import { mapOrganisaatioToSelect, organisaatioSelectMapper } from '../../../tools/organisaatio';
+
+type TSProps = {
+    siirraOrganisaatio: SiirraOrganisaatioon;
+    handleChange: (props: SiirraOrganisaatioon) => void;
+    organisaatio: Organisaatio;
+    organisaatioRakenne: ResolvedRakenne;
+};
+
+export default function TSBody({ siirraOrganisaatio, handleChange, organisaatio, organisaatioRakenne }: TSProps) {
+    const { i18n, language } = useContext(LanguageContext);
+    const targetType =
```

Comment:
maybe good candidate for optional chaining? `organisaatioRakenne?.moveTargetType`

Reply:
true, but not needed, since organisaatoiRakenne is always defined as is moveTargetType. this was a remainder from before refactoring